### PR TITLE
fix: Removed markdown solution

### PR DESCRIPTION
**Pull request by aashishvanand**
Fix for issue 2862

### CODE REVIEW

**Comment by BethanyG** [comment]
Hi @aashishvanand 👋🏽 

Thanks for making the effort to submit a PR.  Unfortunately, per my comments in [Issue 2862](https://github.com/exercism/python/issues/2862), I will be closing this.

**Comment by kotp** [comment]
We reopened this after some thought.  Perhaps you could help us out by providing a docstring or comment at the top of that exercise file giving some information that this is indeed not the solution, but is meant to be refactored?  Even though the directions state this, for some reason there was something missing that prompted this PR, and we can use this PR to prevent this from happening for another student of the track!

**Comment by BethanyG** [comment]
@aashishvanand  -- on second thought, I am re-opening this.  Rather than delete the code to be refactored, it would be very  helpful if you added a [module docstring](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) to the stub file that contains the code, explaining why the code is there, and what the exercise is intended to do.  Something along the lines of:

```python

"""Refactor this code.

The intent of this exercise is to have you practice a refactor of the code in this file
 to increase its readability and maintainability.

We suggest you make notes for reviewers as you do so, 
to give them a starting point for discussion.
"""
```

At the top of the file.   But please also restore the code.  😄   _many thanks!_

**Review by BethanyG — CHANGES_REQUESTED** [review]
1.  Restore code
2.  Add module docstring.

**Comment by aashishvanand** [comment]
@BethanyG if I add module docstring will that be fine?

**Comment by kotp** [comment]
I am not the BethanyG but please do, @aashishvanand .

**Comment by aashishvanand** [comment]
@kotp can you please verify the latest commit if that would be fine? and is that's what you are expecting?

File: exercises/practice/markdown/markdown.py [hunk 1]
```diff
@@ -1,77 +1,9 @@
...
-        res += '</ul>'
-    return res
+    """ Markup
+    Input: would be a markdown phrase for an example
+    Output: expected to pass a HTML equivalent for the same
+    Sample: "#exercism.org is a useful practice ground"
+    you syntax should return <h1>exercism.org is a useful practice ground </h1>
+    Hint: Try exporing regular expressions. Makes life simple
+    """
+    pass
```
Comment (kotp) [inline exercises/practice/markdown/markdown.py:9]: This is missing an `EOL`.  You can append `0x0a` or `U+000A` so the last line is indeed a line.

**Review by kotp — COMMENTED** [review]
I will look at this more and I expect BethanyG will have some thoughts as well. In the meantime, I would ask you to review what is written here and see if it is the best information you can come up with for this.

Like the "input" line shows me input that shows a phrase, but I would expect, in this case that that output would not be so different from that input line.

so we might instead show something on the input line likewhat you wrote for the sample line, and for the output line more like the HTML portion that you showed.

```
Input: exercism.org is a useful practice ground
Output: <h1>exercism.org is a useful practice ground</h1>
```

Though I also think that is not correct as well.  But at least it is not as surprising as what I saw the input and output lines show currently.

**Comment by kotp** [comment]
> @kotp can you please verify the latest commit if that would be fine? and is that's what you are expecting?

I believe the code needs to be restored, and the docstring added.  So this is looking like it is in the midst of a change, not as it should be.

Remember this is a refactoring exercise.  So the "solution" should exist, but it is not the final solution.  The docstring should instead be added to help remind the student of this.

Indeed, restore the branch to what is at `origin/main` and add the docstring that BethanyG suggested located in [this message][1] and that should be good for this.

Also, the title of this PR should be changed, as it really is not as you first imagined it. Perhaps `Fix: Add docstring reminder that this is a refactoring exercise.`

[1]: https://github.com/exercism/python/pull/2863#issuecomment-1017924453

**Comment by aashishvanand** [comment]
Okay, The original solution without making a single change in the editor the code submitted worked fine and passed all the test cases. 
I have removed some cases in the code and have added comments for the user to fill in. So it becomes like an exercise. 
@kotp can you review it now? Let me know if this will be fine.

**Review by kotp — CHANGES_REQUESTED** [review]
Pause and re-read the discussion from the beginning.

This is a refactoring exercise, the code should stay as it is.  Add the docstring that BethanyG proposed, and do not change or remove the code that is meant to be refactored.

This should be the commit that exists here.  Feel free to make the change and force push that change to this branch.

**Comment by BethanyG** [comment]
@aashishvanand -- I think you misunderstood what @kotp is asking you to do.  We did not want you to erase the original code or make **any** edits to it.   We'd like you to **revert** all of the changes you have made up to this point, and restore the code that was in the file before you erased/edited it the first time.  The code should look **exactly like** [this file](https://github.com/exercism/python/blob/main/exercises/practice/markdown/markdown.py).

Once that is done, we're asking you to add the following to the top of the file (_as was noted in my comment above_):

```python
"""Refactor this code.

The intent of this exercise is to have you practice a refactor of the code in this file
 to increase its readability and maintainability.

We suggest you make notes for reviewers as you do so, 
to give them a starting point for discussion.
"""
```

We would rather you not point students at RegEx -- we don't want to over-specify what a potential solution for the exercise might look like.

Additionally, KOTP would like you to change the title of the PR to reflect that you are adding a docstring to the file. 

If it is not possible for you to follow those instructions, I would rather that you simply close this PR.  Thanks.

**Comment by kotp** [comment]
Also, as a note, issue #2862 is specifically set as a "wont fix" and so that should be further indication that the code itself does not need to be changed.  But we will accept a docstring.  This PR still has a life, it is just not as complicated as it is becoming.

**Comment by aashishvanand** [comment]
@BethanyG @kotp sorry for the confusion then. Probably I understood it wrong from the beginning. I have reverted it to the original code with just the additional markup. Also, remove the hints so that they don't confuse the students to stress about Regex.

File: exercises/practice/markdown/markdown.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 
-
 def parse(markdown):
+
+    """ Markup
+    Input: would be a markdown phrase for an example
+    Output: expected to pass a HTML equivalent for the same
+    Sample: "#exercism.org is a useful practice ground"
+    you syntax should return <h1>exercism.org is a useful practice ground </h1>
+    """
```
Comment (kotp) [inline exercises/practice/markdown/markdown.py:10]:
```suggestion
    """Refactor this code.
    
    The intent of this exercise is to have you practice a refactor of the code in this file
    to increase its readability and maintainability.
    
    We suggest you make notes for reviewers as you do so, 
    to give them a starting point for discussion.
    """
```

This is the same suggestion as @BethanyG gave (though to be fair, she did say "something like" and that suggested a bit more creativity could be involved than probably was wanted).

This is would be preferable, though I will defer to you as to the indentation and particular placement.

**Review by kotp — COMMENTED** [review]
Much better, a lot closer to what is needed.

**Comment by kotp** [comment]
There are 4 commits here and there should probably only be one.  If you would like to explore fixing the history that is great.  Otherwise we can do so on our side, when it is ready to come in.

File: exercises/practice/markdown/markdown.py [hunk 3]
```diff
@@ -74,4 +80,4 @@ def parse(markdown):
         res += i
     if in_list:
         res += '</ul>'
-    return res
+    return res
```
[the remark this answers is not in the document]
Comment (aashishvanand) [reply exercises/practice/markdown/markdown.py:83]: Yes will add a blank line

File: exercises/practice/markdown/markdown.py (hunk 2 above)
Comment (aashishvanand) [reply exercises/practice/markdown/markdown.py:10]: Yes, I have made it more readable and have added context to it.

**Comment by BethanyG** [comment]
@aashishvanand @kotp -- I am going to close this PR and add in the docstring myself.  I think there's been too much discussion and confusion around this proposed change, and things have gotten very muddled.  @aashishvanand -- happy to entertain future PRs from you, just .. not this one.

**Comment by kotp** [comment]
Thank you @aashishvanand for the work and exploration with this.

**Pull request closed without merging**